formats.go

```diff
@@ -67,7 +67,38 @@ func writeEnvFile(filename string, data map[string]interface{}, mode os.FileMode
 	return writeFile(filename, buf.Bytes(), mode)
 }
 
+func writeCAChain(filename string, data map[string]interface{}, mode os.FileMode) error {
+	const element = "ca_chain"
+	const suffix = "ca"
+
+	// the chain should be a slice so assert that the type is []interface
+	chain, ok := data[element].([]interface{})
+	if !ok {
+		glog.Errorf("didn't find the certification option: %s", element)
+	}
```

Comment:
Should you not return an error at this point, instead of keeping going? Or if this functionality is optional, you could return nil with a brief comment stating you are aborting early because there is nothing to do? As things stand the code will continue and try to write to an empty filename?

Reply:
Good point, updated it to return nil. There may be use-cases where this element doesn't exist in which case we shouldn't write an empty file.